PINCache/PINOperationQueue.m

```diff
@@ -142,7 +133,27 @@ - (void)dealloc
 - (void)cancelOperation:(id <PINOperationReference>)operationReference
 {
   [self lock];
-    [_canceledOperations addObject:operationReference];
+    PINOperation *operation = [_referenceToOperations objectForKey:operationReference];
+    if (operation) {
+      for (NSMutableOrderedSet *queue in _prioritizedOperationQueues) {
```

Comment:
Can we have some kind of connection operation -> queue so we don't have to iterate through all queues to remove the operation?